src/frontend/src/lib/components/transactions/TransactionContactCard.svelte

```diff
@@ -19,38 +22,55 @@
 
 	const { type, to, from, toExplorerUrl, fromExplorerUrl }: Props = $props();
 
+	let address: string | undefined = $derived(
+		type === 'send' && nonNullish(to) ? to : type !== 'send' && nonNullish(from) ? from : undefined
```

Comment:
The PR is a refactoring but, the logic isn't exactly the same. For example, the address can now be set with `to` even if `from` is nullish which was not the case before. Is that expected @daviddecentage ?